src/components/layout/header/Header.tsx

```diff
@@ -94,51 +85,74 @@ const DesktopMenu: React.FC<MenuProps> = ({ routerPath }): JSX.Element => {
-            key={menuItem.name}
-            onClick={handleClick(menuItem)}
+            className={`okp4-nemeton-web-header-links-mobile-burger-menu ${
+              menuOpen ? 'rotate-down' : 'rotate-up'
+            }`}
           >
-            {menuItem.hash ? (
-              <Link
-                href={`${menuItem.path}${menuItem.hash}`}
-              >
+            <Image
+              alt="mobile-menu"
+              height={40}
+              onClick={toggleMenu}
+              src="/icons/menu-dark.svg"
+              width={40}
+            />
+          </div>
+        )}
+        <div
+          className={classNames('okp4-nemeton-web-header-menu-links', {
+            open: menuOpen
+          })}
+        >
+          {menu.map((menuItem: Route, index: number) => (
+            <div
+              className="okp4-nemeton-web-header-menu-item"
+              key={menuItem.name}
+              onClick={handleClick(menuItem)}
+            >
+              {isMobileScreen && index >= 0 && separatorElement}
+              {menuItem.hash ? (
+                <Link href={`${menuItem.path}${menuItem.hash}`}>
+                  <h2
+                    className={classNames('link-label', {
+                      active:
+                        (routerPath === menuItem.path ||
+                          routerPath?.substring(1) === menuItem.hash) &&
+                        !selectedMenu
+                    })}
+                  >
+                    {menuItem.name}
+                  </h2>
+                </Link>
+              ) : (
                 <h2
                   className={classNames('link-label', {
-                    active:
-                      (routerPath === menuItem.path ||
-                        routerPath?.substring(1) === menuItem.hash) &&
-                      !selectedMenu
+                    active: isActiveMenu(menuItem)
                   })}
                 >
                   {menuItem.name}
                 </h2>
-              </Link>
-            ) : (
-              <h2
-                className={classNames('link-label', {
-                  active: isActiveMenu(menuItem)
-                })}
-              >
-                {menuItem.name}
-              </h2>
-            )}
-          </div>
-        ))}
+              )}
+              {isMobileScreen && selectedMenu?.name === menuItem.name && menuItem.subMenu && (
+                <SubMenuMobile routerPath={routerPath} selectedMenu={selectedMenu} />
+              )}
+              {isMobileScreen &&
+                index === menu.length - 1 &&
+                selectedMenu?.name !== menuItem.name &&
+                separatorElement}
+            </div>
+          ))}
+        </div>
       </div>
-      <SubMenuDesktop
-        routerPath={routerPath}
-        selectedMenu={selectedMenu}
-      />
+      {!isMobileScreen && <SubMenuDesktop routerPath={routerPath} selectedMenu={selectedMenu} />}
```

Comment:
This could be simplified with one single ternary to prevent from multiple assertions.